core/zcncrypto/bls0chain_test.go

```diff
@@ -219,108 +72,82 @@ func TestRecoveryKeys(t *testing.T) {
-	err = verifyScheme.SetPublicKey(pk)
+	err = verifyScheme.SetPublicKey(primaryKey)
 	if err != nil {
 		t.Fatalf("Set public key failed")
 	}
-	if ok, err := verifyScheme.Verify(sig1, hash); err != nil || !ok {
-		fmt.Println("err", err)
+	if ok, err := verifyScheme.Verify(addSig, hash); err != nil || !ok {
 		t.Fatalf("Verification failed\n")
 	}
 }
 
 func TestSplitKey(t *testing.T) {
-	// Generate 0th signature based on this primaryKey.
-	primaryKeyStr := `212ba4f27ffffff5a2c62effffffffcdb939ffffffffff8a15ffffffffffff8d`
-
-	scheme0 := NewBLS0ChainScheme()
-	err := scheme0.SetPrivateKey(primaryKeyStr)
+	primaryKeyStr := `c36f2f92b673cf057a32e8bd0ca88888e7ace40337b737e9c7459fdc4c521918`
+	sig0 := NewBLS0ChainScheme()
+	err := sig0.SetPrivateKey(primaryKeyStr)
 	if err != nil {
 		t.Fatalf("Set private key failed - %s", errors.Top(err))
 	}
 	hash := Sha3Sum256(data)
-	sig0, err := scheme0.Sign(hash)
+	signature, err := sig0.Sign(hash)
 	if err != nil {
 		t.Fatalf("BLS signing failed")
 	}
-
-	// Split keys.
 	numSplitKeys := int(2)
-	w, err := scheme0.SplitKeys(numSplitKeys)
+	w, err := sig0.SplitKeys(numSplitKeys)
 	if err != nil {
 		t.Fatalf("Splitkeys key failed - %s", errors.Top(err))
 	}
-
-	// Generate schemes from the split keys.
 	sigAggScheme := make([]BLS0ChainScheme, numSplitKeys)
 	for i := 0; i < numSplitKeys; i++ {
 		sigAggScheme[i].SetPrivateKey(w.Keys[i].PrivateKey)
 	}
-
-	// Aggregate the signatures generated by each split key.
 	var aggrSig string
 	for i := 1; i < numSplitKeys; i++ {
-		tmpSig, err := sigAggScheme[i].Sign(hash)
-		if err != nil {
-			fmt.Println("err", err)
-			t.Fatalf("Shouldn't have gotten error with Sign()")
-		}
-
-		aggrSig, err = sigAggScheme[0].Add(tmpSig, hash)
-		if err != nil {
-			fmt.Println("err", err)
-			t.Fatalf("Shouldn't have gotten error with Add()")
-		}
+		tmpSig, _ := sigAggScheme[i].Sign(hash)
+		aggrSig, _ = sigAggScheme[0].Add(tmpSig, hash)
 	}
-
-	if aggrSig != sig0 {
+	if aggrSig != signature {
 		t.Fatalf("split key signature failed")
 	}
 }
 
-// TODO: this gosdk's MiraclToHerumiPK function needs to be replaced wherever
-// it is used with a local version, so that gosdk is able to compile without
-// C++ dependencies.
-//
-// func TestMiraclToHerumiPK(t *testing.T) {
-// 	miraclpk1 := `0418a02c6bd223ae0dfda1d2f9a3c81726ab436ce5e9d17c531ff0a385a13a0b491bdfed3a85690775ee35c61678957aaba7b1a1899438829f1dc94248d87ed36817f6dfafec19bfa87bf791a4d694f43fec227ae6f5a867490e30328cac05eaff039ac7dfc3364e851ebd2631ea6f1685609fc66d50223cc696cb59ff2fee47ac`
-// 	pk1 := MiraclToHerumiPK(miraclpk1)
-//
-// 	require.EqualValues(t, pk1, "68d37ed84842c91d9f82389489a1b1a7ab7a957816c635ee750769853aeddf1b490b3aa185a3f01f537cd1e9e56c43ab2617c8a3f9d2a1fd0dae23d26b2ca018")
-//
-// 	// Assert DeserializeHexStr works on the output of MiraclToHerumiPK
-// 	var pk bls.PublicKey
-// 	err := pk.DeserializeHexStr(pk1)
-// 	require.NoError(t, err)
-// }
+func TestMiraclToHerumiPK(t *testing.T) {
```

Comment:
I think `TestMiraclToHerumiPK` is not used outside gosdk? I remember blobber has its own `TestMiraclToHerumiPK`. It's fine to commented it.

Reply:
leave it first. we still need to migrate HermuiBLS to Miracl. Let's clean all of them once it is migrated.